src/fe-common/core/fe-common-core.c

```diff
@@ -462,25 +462,73 @@ void fe_common_core_finish_init(void)
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
 	g_return_val_if_fail(array != NULL, FALSE);
+	WI_ITEM_REC *item = window_item_find_window(dest->window, dest->server, dest->target);
+	int channel_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "CHANNEL");
+	int query_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "QUERY");
+	
+	char **tmp;
+	for (tmp = array; *tmp != NULL; tmp++) {
+		char *tagtarget = NULL;
+		if (g_ascii_strcasecmp(*tmp, "*") == 0)
+			return TRUE;
+		if (g_ascii_strcasecmp(*tmp, dest->target) == 0)
+			return TRUE;
 
-	if (strarray_find(array, "*") != -1)
-		return TRUE;
+		if (dest->server_tag != NULL) {
```

Comment:
I think that you can just `return FALSE` if `server_tag == NULL` since without that you can't really match anything (beside the "bare" `#`, `=` and `@`) and it'd allows to simplify the following code quite a bit.
Better ask @ailin-nemui @dequis